app/scripts/lib/account-tracker.js

```diff
@@ -70,22 +66,27 @@ export default class AccountTracker {
     };
 
     this._provider = opts.provider;
-    this._query = pify(new EthQuery(this._provider));
     this._blockTracker = opts.blockTracker;
-    // blockTracker.currentBlock may be null
-    this._currentBlockNumber = this._blockTracker.getCurrentBlock();
-    this._blockTracker.once('latest', (blockNumber) => {
-      this._currentBlockNumber = blockNumber;
-    });
+
     // bind function for easier listener syntax
     this._updateForBlock = this._updateForBlock.bind(this);
     this.getCurrentChainId = opts.getCurrentChainId;
+    this.getNetworkClientById = opts.getNetworkClientById;
     this.getNetworkIdentifier = opts.getNetworkIdentifier;
     this.preferencesController = opts.preferencesController;
     this.onboardingController = opts.onboardingController;
 
+    // blockTracker.currentBlock may be null
+    const currentChainId = this.getCurrentChainId();
+    this._currentBlockNumberByChainId = {
+      [currentChainId]: this._blockTracker.getCurrentBlock(),
+    };
+    this._blockTracker.once('latest', (blockNumber) => {
+      this._currentBlockNumberByChainId[currentChainId] = blockNumber;
+    });
```

Comment:
Does this logic still work since `currentChainId` will always be the same as the chainId when the controller is instantiated and the `_blockTracker` will not always be tied to that chain?

Reply:
good catch! https://github.com/MetaMask/metamask-extension/pull/21868/commits/b62526e1a34e3532ef46e586839b704054819010